include/core/trace.h

```diff
@@ -540,6 +541,7 @@ typedef struct trace_t trace_t;
 
 #define start_trace(...)
 #define stop_trace(...)
+#define stop_trace_locked(...)
```

Comment:
@erlingrj added this `stop_trace_locked` in `enclaves3`, but I'm not sure why. The only place it is called seems to be in the `termination` function, and it seems to be called without holding any locks.  Maybe we should just have `stop_trace` always acquire the lock?  That said, maybe the reason is that acquiring locks during termination is risky because it can prevent a program from exiting that has deadlocked on these same locks.  Was this the reason for this change?  If so, I guess probably all we need is a comment in the `termination` function explaining why this is called instead of `stop_trace`.

Reply:
I'm going to mark this resolved and open an issue because I don't think this is important to block merging this.